e2e/framework/cluster.go

```diff
@@ -0,0 +1,71 @@
+package framework
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// Cluster represents an interface to test bootkube based kubernetes clusters.
+// The simplest way to write tests against it is to just write go tests that call
+// out to NewCluster at the beginning of each test. External tooling should first
```

Comment:
It seems more uniform to make a call to the framework to get a client at every function then sharing globals in testing packages. No reason that NewCluster can't just be GetCluster that returns a global from the framework package assuming `Clientset` is safe for parallel use (docs didn't make this obvious if true). Its possible certain tests could be safe to run in parallel in the future.